js/packages/web/src/actions/getStorefront.ts

```diff
@@ -76,13 +81,25 @@ const fetchFromSource = async (
 
       return acc;
     }, {});
-    let pubkeyDenyList: Array<String> = [];
+    let pubkeydenyListCacheKey: Array<String> = [];
     try {
-      pubkeyDenyList = await fetchDenyList();
+      const cachedValue = await redisClient.get(denyListCacheKey);
+      pubkeydenyListCacheKey = JSON.parse(cachedValue) || [];
+
+      if (!pubkeydenyListCacheKey.length) {
+        pubkeydenyListCacheKey = await fetchdenyListCacheKey();
+        // @ts-ignore
+        redisClient.set(
+          denyListCacheKey,
+          JSON.stringify(pubkeydenyListCacheKey.result),
+        );
+        redisClient.expire(denyListCacheKey, 3600); // cache the entire list for an hour
```

Comment:
Probably only need to cache for like ~5min, would make for snappier ban responses.